core/src/main/java/org/testcontainers/utility/PrefixingImageNameSubstitutor.java

```diff
@@ -0,0 +1,49 @@
+package org.testcontainers.utility;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.NoArgsConstructor;
+import org.testcontainers.UnstableAPI;
+
+/**
+ * An {@link ImageNameSubstitutor} which applies a prefix to all image names, e.g. a private registry host and path.
+ * The prefix may be set via an environment variable (<code>TESTCONTAINERS_IMAGE_NAME_PREFIX</code>) or an equivalent
+ * configuration file entry (see {@link TestcontainersConfiguration}).
+ */
+@UnstableAPI
+@NoArgsConstructor
+public class PrefixingImageNameSubstitutor extends ImageNameSubstitutor {
```

Comment:
Suggestion: let's add a warning that this class is public only because Java's `ServiceLoader` requires it to be public, similar to:
https://github.com/reactor/reactor-core/blob/246251727bb5020c21abc1d34847d29afd8c7ec3/reactor-core/src/main/java/reactor/core/scheduler/ReactorBlockHoundIntegration.java#L27-L28

Reply:
Removed from this PR